csharp_dotnetcore/14.NLP-With-Dispatch/NlpDispatch/BotServices.cs

```diff
@@ -0,0 +1,68 @@
+{
+    /// <summary>
+    /// Represents the bot's references to external services.
+    ///
+    /// For example, Application Insights, Luis models and QnaMaker services
+    /// are kept here (singletons).  These external services are configured
+    /// using the BotConfigure class (based on the contents of your ".bot" file).
+    /// </summary>
+    [Serializable]
+    public class BotServices
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BotServices"/> class.
+        /// </summary>
+        /// <param name="client">An Application Insights <see cref="TelemetryClient"/> instance.</param>
+        /// <param name="qnaServices">A dictionary of named <see cref="QnAMaker"/> instances for usage within the bot.</param>
+        public BotServices(TelemetryClient client, Dictionary<string, QnAMaker> qnaServices, Dictionary<string, LuisRecognizer> luisServices)
+        {
+            TelemetryClient = client ?? throw new ArgumentNullException(nameof(client));
+            QnAServices = qnaServices ?? throw new ArgumentNullException(nameof(qnaServices));
+            LuisServices = luisServices ?? throw new ArgumentNullException(nameof(luisServices));
+        }
+
+        /// <summary>
+        /// Gets the Application Insights Telemetry client.
+        /// Use this to log new custom events/metrics/traces/etc into your
+        /// Application Insights service for later analysis.
+        /// </summary>
+        /// <value>
+        /// The Application Insights <see cref="TelemetryClient"/> instance created based on configuration in the .bot file.
+        /// </value>
+        public TelemetryClient TelemetryClient { get; }
+
+        /// <summary>
+        /// Gets the (potential) set of QnA Services used.
+        /// Given there can be multiple QnA services used in a single bot,
+        /// QnA is represented as a Dictionary.  This is also modeled in the
+        /// ".bot" file since the elements are named (string).
+        /// This sample only uses a single QnA instance.
+        /// </summary>
+        /// <value>
+        /// A QnAMaker client instance created based on configuration in the .bot file.
+        /// </value>
+        public Dictionary<string, QnAMaker> QnAServices { get; } = new Dictionary<string, QnAMaker>();
+
+        /// <summary>
+        /// Gets the (potential) set of Luis Services used.
+        /// Given there can be multiple Luis services used in a single bot,
+        /// LuisServices is represented as a Dictionary.  This is also modeled in the
+        /// ".bot" file since the elements are named (string).
+        /// This sample only uses a single Luis instance.
+        /// </summary>
+        /// <value>
+        /// A <see cref="LuisRecognizer"/> client instance created based on configuration in the .bot file.
+        /// </value>
+        public Dictionary<string, LuisRecognizer> LuisServices { get; } = new Dictionary<string, LuisRecognizer>();
```

Comment:
>Dictionary [](start = 15, length = 10)

Same comment applies for discussion.

Reply:
Leaving open in case there is some action item from the comment above.